### DISCUSSION TITLE
Read numeric data files in MIMIC-IV Waveform Database v0.1.0

### DISCUSSION THREAD

**Discussion by fwhiemstra** [MIT-LCP/mimic-code#1392] [MIMIC-IV]
I want to read the numeric data files in MIMIC-IV Waveform Database v0.1.0 in Python. For as far as I understood, this is not possible with the wfdb package. The code provided on https://physionet.org/content/mimic4wdb/0.1.0/ requires downloading of the .csv.gz file. Is there a way to read the numeric data stored in MIMIC-IV Waveform Database without having to download it?

**Comment by tompollard** [comment]
Hi @fwhiemstra. Right now, it is necessary to download the waveforms in some way before analysis, but this could be done with a function call directly from the WFDB package (in a cloud notebook, such as Google Colab if desired!).

Please take a look at https://wfdb.io/mimic_wfdb_tutorials for some executable notebooks that demonstrate how to load and explore data using the WFDB python package. Do these help?

**Reply by pavanderkleij** [reply]
Hi @tompollard , the WFDB-package doesn't seem to support the file extension .csv.gz with any of the given functions. The closest option i found was the function wfdb.rdann, meant for annotations, where the file extension can be specified. Unfortunately, since the function isn't meant for numerical files, it errors during reshaping of the data.

Have you got any suggestions on how to implement the WFDB-package to load the .csv.gz data without manually downloading it each time? 

Kind regards, Paul

**Comment by fwhiemstra** [comment]
Hi @tompollard, 
Thanks for your reaction! The tutorial is really helpful on how to load the waveform data with the WFDB python package, but it is still not clear how to load the numeric files. Is this possible without having to download the .csv.gz file on your computer?

Thank you in advance!
